GamepadButtonAndAxisEvents/explainer.md

```diff
@@ -0,0 +1,201 @@
+
+### Goals
+
+Reduce input latency by moving away from constant polling and introducing event-driven input handling. 
+
+### Non-goals
+
+The existing polling mechanism will not be deprecated at all. We are just proposing an alternative way of handling input events and applications are free to select whichever they prefer.
+
+## User research
+
+Research and experimental implementations in Firefox suggest that event-driven input handling can significantly improve responsiveness and efficiency. Based on this research, the following challenges with prior implementations have been identified and addressed:
+
+- Inadequate handling of analog button states: Previous implementations only triggered events when the button's pressed state changed, without accounting for changes in the button's numeric value or "touched" state. The new proposal ensures that all relevant state changes, including value adjustments and touch states, are captured, providing more comprehensive input data.
+
+- Potential loss of gamepad state before event listeners execute: The gamepad state at the time the event was created may be lost, as the event includes a reference to the live Gamepad object, which may have updated its state before the event listener is invoked. For instance, if a button is quickly pressed and released, the Gamepad's button state may already indicate that the button is unpressed when the event listener is called. By combining polling with event-driven updates, the new approach mitigates this issue by ensuring the most recent state is captured, and events are triggered only when significant changes occur, reducing the risk of missing relevant input data.
+
+- Unintuitive event targets: Previous event systems fired on the global window object, which was counterintuitive. The windows object is not directly tied to a specific gamepad instance and since multiple gamepads can be connected, handleing events at the gamepad level will be more logical. Most input APIs like KeyboardEvent and MouseEvent dispatch events on the relevant input device not the global window. Gamepad events on window break this consistency. With the new proposal, events will be triggered directly on the Gamepad object that generated the event, providing a clearer and more developer-friendly experience.
+
+- Event spam from per-frame axis updates:When a separate event for each changed axis on each frame of input is fired, if events are received too rapidly then they will queue, introducing input latency. 
+
+## Proposed Approach
```

Comment:
We need to add sample code that shows usage of the new APIs.  The current examples do not show how web developers would use these APIs.

For example:

https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/IndexedDbGetAllEntries/explainer.md#key-scenarios

See also:

Writing Effective Explainers: https://w3ctag.org/explainers/, which says "Practical Use Cases: Show example pieces of code that use the proposed solution to solve the user-facing problems."